MapboxDirections/MBIntersection.swift

```diff
@@ -0,0 +1,109 @@
+import Foundation
+
+/**
+ A single cross street along a step.
+ */
+@objc(MBIntersection)
+public class Intersection: NSObject, NSSecureCoding {
+    
+    /**
+     Index into bearings/entry array.
+     
+     The index of the item in the headings array that corresponds to the road that the containing route step uses to approach the intersection.
+    */
+    public var approachIndex: Int
+    
+    /**
+     Index into the bearings/entry array.
+     
+     The index of the item in the headings array that corresponds to the road that the containing route step uses to leave the intersection.
+    */
+    public var outletIndex: Int
+    
+    /**
+     An array of booleans, corresponding in a 1:1 relationship to the headings.
+     
+     A value of true indicates that the respective road could be entered on a valid route. false indicates that the turn onto the respective road would violate a restriction.
+     */
+    public let entry: [Bool]
```

Comment:
Still waiting for this method to be renamed to `outlets` and changed to be an IndexSet.